### PR TITLE
Make component-library analytics listener site-wide

### PR DESCRIPTION
**Pull request by bsmartin-ep**
As discussed, we need to move our CustomEvent (`component-library-analytics`) listener from the static-pages app to something site wide.

Taking the lead from [platform/site-wide/wysiwyg-analytics-setup.js](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/site-wide/wysiwyg-analytics-setup.js), I moved the listener and lookup table over.

Let me know what you think!

### CODE REVIEW

File: src/platform/site-wide/tests/component-library-analytics-setup.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  event: 'int-modal-click',
+  'modal-title': 'Modal title',
+  'modal-status': 'info',
+  'modal-primaryButtonText': 'Button Text 1',
+  'modal-secondaryButtonText': 'Button Text 3',
+};
+
+subscribeComponentAnalyticsEvents(event, recordEvent);
+
+expect(recordEvent.calledWith(dataLayerEvent)).to.be.true;
```
Comment (ncksllvn) [inline src/platform/site-wide/tests/component-library-analytics-setup.unit.spec.js:29]: Oh sorry, my code was untested and incomplete. I was just trying to show a PoC for getting started but it is up to you as far as what gives you confidence. We'll at least need to wrap the code in `describe` and `it` Mocha methods

Comment (bsmartin-ep) [reply src/platform/site-wide/tests/component-library-analytics-setup.unit.spec.js]: No worries - I changed it to `calledWith` and it's very close to the type of testing we're doing on the component-library side of things.

Updated with _describe_ / _it_.

File: src/platform/site-wide/tests/component-library-analytics-setup.unit.spec.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      event: 'int-modal-click',
+      'modal-title': 'Modal title',
+      'modal-status': 'info',
+      'modal-primaryButtonText': 'Button Text 1',
+      'modal-secondaryButtonText': 'Button Text 3',
+    };
+
+    subscribeComponentAnalyticsEvents(event, recordEvent);
+
+    expect(recordEvent.calledWith(dataLayerEvent)).to.be.true;
```
Comment (ncksllvn) [inline src/platform/site-wide/tests/component-library-analytics-setup.unit.spec.js:31]: Great assertion 👍

**Review by ncksllvn — APPROVED** [review]
LGTM

**Review by jonwehausen — APPROVED** [review]

**Review by bkjohnson — APPROVED** [review]
Looks good :+1:

**Review by alexandec — APPROVED** [review]

**Pull request merged**